experimental/yarpl/include/reactivestreams/ReactiveStreams.h

```diff
@@ -0,0 +1,39 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#pragma once
+
+#include <exception>
+#include <memory>
+
+namespace reactivestreams_yarpl {
+// TODO remove 'yarpl' if we agree on this usage elsewhere
+
+class Subscription {
+ public:
+  virtual ~Subscription() = default;
+  virtual void request(int64_t n) = 0;
+  virtual void cancel() = 0;
+};
+
+template <typename T>
+class Subscriber {
+ public:
+  virtual ~Subscriber() = default;
+  virtual void onSubscribe(Subscription*) = 0;
```

Comment:
In the general case, Subscription owns Subscriber, and will delete it when it finishes. (This is a consequence of cancel() not providing a notification to the subscriber, per the core spec. Thus, only the subscription knows when to clean up, never the subscriber.) If a class that has both interfaces is defined, it should manage the lifetime the same way: the subscription manages the cleanup; if there is no separate subscriber object, there's nothing to delete.